lib/driver.js

```diff
@@ -412,10 +418,25 @@ class XCUITestDriver extends BaseDriver {
       await this.wda.uninstall();
       this.logEvent('wdaUninstalled');
     } else if (!util.hasValue(this.wda.webDriverAgentUrl) && (await this.wda.isRunning())) {
-      log.info(`Will reuse previously cached WDA instance at '${this.wda.url.href}'. ` +
-               `Set the wdaLocalPort capability to a value different from ${this.wda.url.port} ` +
-               `if this is an undesired behavior.`);
-      this.wda.webDriverAgentUrl = this.wda.url.href;
+      // Old WDA has no bundleId in the /status. Then we only can obtain `null` as the currentAppBundleId.
+      // Thus, `isRunning()` is the only way to confirm if WDA is running in the target device.
+      const currentAppBundleID = (await this.wda.getRunningBundleId());
+      if (util.hasValue(currentAppBundleID) && (util.hasValue(this.opts.updatedWDABundleId)) && (this.opts.updatedWDABundleId !== currentAppBundleID)) {
+        log.info(`Will uninstall running WDA since it has different bundle id. The actual value is '${currentAppBundleID}'.`);
+        await this.wda.uninstall();
+      } else if (util.hasValue(currentAppBundleID) && (WDA_RUNNER_BUNDLE_ID !== currentAppBundleID)) {
+        log.info(`Will uninstall running WDA since it isn't default, '${WDA_RUNNER_BUNDLE_ID}', bundle id.`);
```

Comment:
doesn't the condition above cover this case?

Reply:
No, the 1st condition doesn't cover this 2nd case.

1. A user sets `updatedWDABundleId: "com.example.WebDriverAgent"` and runs tests
2. The user removes `updatedWDABundleId` and runs another tests
    - `this.opts.updatedWDABundleId` is `undefined`. `currentAppBundleID` is "com.example.WebDriverAgent".

In the case, WDA which will run in the test device should be WDA_RUNNER_BUNDLE_ID since the user doesn't set `updatedWDABundleId`.

Reply:
Ah, sorry. This was my wrong. (My previous wrong case's one)